### ISSUE TITLE
Narrator reads information of "0th" bit button incorrectly

### ISSUE THREAD

**Issue by MicrosoftIssueBot** [microsoft/calculator#385]
**Repro Steps**
* Launch "Calculator" app. Launch narrator.
* Navigate to Menu and select "Programmer".
* Navigate to “Bit toggling keypad” radio button and select it.
* Navigate to "0th" bit button and observe the narrator announcement. 

**Actual Result**
Narrator reads "0 and th" separately for 0th bit button. 

**Expected Result**
Narrator should read "0th" bit button on/off.

**Test Environment**
OS Version: Windows 10 Build 18351.1
App Name: Calculator 
App Version: 10.1902.16.0

**Comment by MicrosoftIssueBot** [comment]
This is your friendly Microsoft Issue Bot. I created this issue automatically as requested by a team member.

**Comment by brulecap** [comment]
New to the calculator community and would love to contribute any way I can.

I verified that this is not only a problem with Narrator but also with the NVDA screen reader. Both pronounce "0th" as "0" and "th" separately. Other screen readers?

I then went down a bit of a rabbit hole when considering the propriety/impropriety of using the ordinal term zeroth(0th). It seems the 1st bit would be bit position 0, but considering that we are performing 0 based numbering, I grudgingly accepted the propriety of using zeroth(0th) to denote the first bit.

2 solutions I came up with:

1)Change the name="0" data value from 0th( or language-specific equivalents) to 0(non-language-specific) in all resources.resw locale files.
	The result would be that Narrator would read "0 bit value 0" instead of "0 th bit value 0". All others would stay the same. i.e. "32nd bit value 0".

2)Change the string returned from BitFlipAutomationNameConverter.
	Example: Instead of returning "1st bit value 1" have it return "Bit 1 value 1".
We could use the name contained in "parameter" in the parameter list of BitFlipAutomationNameConverter to achieve this. Should we? The right way would be to change all name data values from their ordinal to cardinal representation in all resources.resw locale files? i.e: value "1" instead of "1st".

I welcome any feedback. I would love to tackle this once a proper resolution is decided upon by the community.

**Comment by danbelcher-MSFT** [comment]
What about "low bit"/"high bit"?

**Comment by grochocki** [comment]
How about just ["least significant bit"](https://en.wikipedia.org/wiki/Bit_numbering)?

**Comment by brulecap** [comment]
@danbelcher-MSFT, not sure how "low bit"/"high bit" could be used. I assume "low bit" would be bit position 0, "high bit" bit position 63. What would be the convention for the remaining 62 bits? Am I missing something?

@grochocki, "least significant bit" would work but departs from the convention used for the remaining 63 bits. 

Both of these would also require providing language specific translations of either "high" and "low", or "least significant" in a number of resources.resw locale files.

My suggestion is solution 1 above. This departs from the current solution very little, "0th bit" ≈ "0 bit" in my mind. Also, this solution does not require language specific translations. Good?

**Comment by grochocki** [comment]
> "least significant bit" would work but departs from the convention used for the remaining 63 bits.

True, but the convention here is self-defined. "Least significant bit" is an industry term for the low bit. We _could_ also make the highest bit "Most significant bit" (also in line with industry terminology). IMO, this is a better solution than "0 bit".

> Both of these would also require providing language specific translations of either "high" and "low", or "least significant" in a number of resources.resw locale files.

We only have to provide the English resource. Our internal localization process will localize the string into all supported languages.

**Comment by brulecap** [comment]
Thanks, @grochocki. Just what I was looking for. I'll make the changes and submit a PR per [guidelines](https://github.com/Microsoft/calculator/blob/master/CONTRIBUTING.md#submitting-a-pull-request-and-participating-in-code-review). I appreciate the opportunity.

**Comment by danbelcher-MSFT** [comment]
I'm going to suggest that we make no change here and instead consider this a Windows Narrator bug.  Narrator recognizes and correctly pronounces "1st, 2nd, 3rd, ...".  It should also correctly handle "0th".

The current proposal also doesn't seem like an appropriate change.  The "most significant bit" depends on the context (63 for Qword but 7 for Byte).  There are other alternatives we could try, such as "zeroth" and "zeroeth".  But we should start with seeing how the Narrator team considers this issue.  In the end, Narrator is driven by a TTS speech and pronunciations don't need to be perfect.

**Comment by danbelcher-MSFT** [comment]
Bug for this has been filed internally and is currently sitting on the TTS team.  Will update when we have more info.  Tracked internally with: 20820966

**Comment by grochocki** [comment]
> Narrator recognizes and correctly pronounces "1st, 2nd, 3rd, ...". It should also correctly handle "0th".

I completely agree! Thanks for filing that bug on TTS.

> I'm going to suggest that we make no change here

While "0th" and "7th" (in the context of a Byte) are not _inaccurate_, they also do not align with industry standard terms for these bits (LSB/MSB, respectively). I think we should use this opportunity to improve accessibility by switching to that terminology.

**Comment by brulecap** [comment]
Well, regardless of how you decide to proceed, I'm still going to continue looking into adding MSB for qword, dword, word, and byte. It has been a good opportunity for me to learn the code base. Still not sure how to do it, but I find a new piece to the puzzle every day.

**Comment by danbelcher-MSFT** [comment]
Whoops!
